src/crypto/equihash.cpp

```diff
@@ -16,6 +16,10 @@
 #include "config/bitcoin-config.h"
 #endif
 
+#if defined(__APPLE__)
+#include "compat/endian.h"
+#endif
```

Comment:
It should be sufficient to include ``compat/endian.h`` unconditionally, I think.